go/internal/initutil/ipfs.go

```diff
@@ -495,11 +494,13 @@ func (m *Manager) setupIPFSConfig(cfg *ipfs_cfg.Config) ([]libp2p.Option, error)
 	}
 
 	// disable original connection manager
-	cfg.Swarm.ConnMgr.Type = "none"
+	cfg.Swarm.ConnMgr.Type = ipfs_cfg.NewOptionalString("none")
 
 	// configure custom conn manager
 	cm, err := connmgr.NewConnManager(
-		cfg.Swarm.ConnMgr.LowWater, cfg.Swarm.ConnMgr.HighWater, connmgr.WithGracePeriod(ipfs_cfg.DefaultConnMgrGracePeriod),
+		int(cfg.Swarm.ConnMgr.LowWater.WithDefault(50)),
+		int(cfg.Swarm.ConnMgr.HighWater.WithDefault(100)),
```

Comment:
Question: This is the code that should prevent the "too many open files error". Correct?